src/exchange_adapters/okx.ts

```diff
@@ -0,0 +1,117 @@
+import { Exchange } from '../utils'
+import { BaseExchangeAdapter, ExchangeAdapter, ExchangeDataType, Ticker, Trade } from './base'
+
+export class OKXAdapter extends BaseExchangeAdapter implements ExchangeAdapter {
+  baseApiUrl = 'https://www.okx.com/api/v5'
+  readonly _exchangeName = Exchange.OKX
+  readonly _certFingerprint256 =
+    '3A:BB:E6:3D:AF:75:6C:50:16:B6:B8:5F:52:01:5F:D8:E8:AC:BE:27:7C:50:87:B1:27:A6:05:63:A8:41:ED:8A'
+
+  private static readonly tokenSymbolMap = OKXAdapter.standardTokenSymbolMap
+
+  protected generatePairSymbol(): string {
+    const base = OKXAdapter.tokenSymbolMap.get(this.config.baseCurrency)
+    const quote = OKXAdapter.tokenSymbolMap.get(this.config.quoteCurrency)
+    return `${base}-${quote}`
+  }
+
+  async fetchTicker(): Promise<Ticker> {
+    const tickerJson = await this.fetchFromApi(
+      ExchangeDataType.TICKER,
+      `market/ticker?instId=${this.pairSymbol}`
+    )
+
+    return this.parseTicker(tickerJson)
+  }
+
+  async fetchTrades(): Promise<Trade[]> {
+    // Trade data is not needed by oracle but is required by the parent class.
```

Comment:
maybe we should add an issue in this sprint to get rid of this across the codebase. the more adaptors we add the more annoying it's going to get

Reply:
agree

Reply:
++